contracts/LRTDepositPool.sol

```diff
@@ -164,13 +177,95 @@ contract LRTDepositPool is ILRTDepositPool, LRTConfigRoleChecker, PausableUpgrad
     }
 
     /// @dev private function to mint primeETH
-    /// @param primeEthAmount Amount of primeETH minted
-    function _mint(uint256 primeEthAmount) private {
+    /// @param amount Amount of primeETH to be minted
+    function _mint(uint256 amount) private {
         address primeETH = lrtConfig.primeETH();
         // mint primeETH for user
-        IPrimeETH(primeETH).mint(msg.sender, primeEthAmount);
+        IPrimeETH(primeETH).mint(msg.sender, amount);
     }
 
+    /*//////////////////////////////////////////////////////////////
+                        Withdraw functions
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice PrimeETH staker requests the withdrawal of a liquid staking token (LST) from
+    // its respective underlying EigenLayer strategy.
+    /// @dev Will emit the `Withdrawal` event from EigenLayer's `DelegationManager` contract
+    /// which is needed for the `claimWithdrawal` call.
+    /// @param asset address of the liquid staking token (LST) being requested. eg OETH. Can not be WETH.
+    /// @param assetAmount the amount of LSTs to withdraw.
+    /// @param maxPrimeETH the maximum amount of primeETH tokens that can be burned.
+    /// @return primeETHAmount the amount of primeETH tokens that were burned.
+    function requestWithdrawal(
+        address asset,
+        uint256 assetAmount,
+        uint256 maxPrimeETH
+    )
+        external
```

Comment:
let's add `requireSupportedAsset` to this (like we use for deposits). This way the asset can't be set to something weird. In theory eigenlayer will revert on non zero amounts of weird assets, but let's just not let it get that far.

Reply:
`requestWithdrawal` will only accept WETH due to this check
```Solidity
// Can only withdraw the supported asset
if (asset != WITHDRAW_ASSET) {
    revert NotWithdrawAsset();
}
```

There is also a zero amount check
```Solidity
if (assetAmount == 0) {
    revert ZeroAmount();
}
```

Reply:
`asset != WITHDRAW_ASSET`

Lol, my bad. This one is all good.